examples/layertree.js

```diff
@@ -100,7 +126,19 @@ app.LayertreeController.prototype.getLayer = function(node) {
  * @export
  */
 app.LayertreeController.prototype.onButtonClick = function(node, layer) {
-  window.alert(node['name'] + ', ' + layer.get('type'));
+  if (!angular.isDefined(node.metadataPromise_)) {
```

Comment:
Yeah… If you really want to show how to do it if the metadata URL is not the same for every layer then I'd rather use a new object rather that storing the promises in the nodes themselves. And anyway, it would be one metadata URL per **layer**, as opposed to per **node**.

So, in the controller constructor:

```js
  /**
   * @type {Object.<string, !angular.$q.Promise>}
   * @private
   */
  this.promises_ = {}

  …
```

And in the `onButtonClick` function:

```js
  var layerType = node['layerType'];
  if (!(layerType in this.promises_)) {
    this.promises_[layerType] = $http.get(…).then(function(resp) {
      // …
      return html;
    });
  }
  var infoPopup = this.infoPopup_;
  this.promises_[layerType].then(function(html) {
    // …
  });
```

Makes sense?